### PR TITLE
chore: add FIPS documentation

### PR DESCRIPTION
**Pull request by arriqaaq**

### CODE REVIEW

File: src/master/production/fips.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# FIPS 140-2
+
+<WrappedSection>
+
+The Federal Information Processing Standard (FIPS) 140-2 publication describes United States government-approved security requirements for cryptographic modules. FIPS-140 series is a collection of computer security standards set by the National Institute of Standards and Technology (NIST) for the United States government. FIPS 140–2 defines the critical security parameters vendors must use for encryption before selling to the U.S government.
```
Comment (tomekkolo) [inline src/master/production/fips.md:5]: is it possible to have some link ?

File: src/master/production/fips.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+For a fully FIPS-compliant deployment of immudb a few things are required:
+
+- immudb must be compiled with a FIPS validated cryptographic module
+- immudb must be configured to use FIPS-approved cryptographic algorithms
+- immudb components (immuadmin and immuclient) must be compiled with a FIPS-validated cryptographic module
+
+For ImmuDB, adherence to FIPS 140-2 is ensured by:
```
Comment (tomekkolo) [inline src/master/production/fips.md:13]: immudb

File: src/master/production/fips.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- Using FIPS approved / NIST-recommended cryptographic algorithms through the use of `goboring/golang` container image. Since the native go crypto standard library is not FIPS compliant, we use the Google-provided Go implementation that has patches on top of standard Go to enable integrating BoringCrypto. ImmuDB components are built with this image as a build base.
```
Comment (tomekkolo) [inline src/master/production/fips.md:14]: ImmuDB -> immudb

**Review by tomekkolo — COMMENTED** [review]
are there any concerns worth to mention when you i.e. want to run old db using fips db version ? So migrate to fips ?

Some words about replication ? If master is in FIPS, replicas also must be in fips or it is irrelevant ?

File: src/master/production/fips.md (hunk 2 above)
Comment (arriqaaq) [reply src/master/production/fips.md:13]: ack

File: src/master/production/fips.md (hunk 3 above)
Comment (arriqaaq) [reply src/master/production/fips.md:14]: ack

**Comment by arriqaaq** [comment]
@tomekkolo I did not quite understand the concerns, you mean to mention how to change older versions into FIPS compliant binaries?

I think we don't have to mention about replication in FIPS, because that is related to deployment setup. Here the information is more about FIPS compliant binaries/images and where to get them from right

File: src/master/production/fips.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+For a fully FIPS-compliant deployment of immudb a few things are required:
+
+- immudb must be compiled with a FIPS validated cryptographic module
+- immudb must be configured to use FIPS-approved cryptographic algorithms
+- immudb components (immuadmin and immuclient) must be compiled with a FIPS-validated cryptographic module
+
+For immudb, adherence to FIPS 140-2 is ensured by:
+- Using FIPS approved / NIST-recommended cryptographic algorithms through the use of `goboring/golang` container image. Since the native go crypto standard library is not FIPS compliant, we use the Google-provided Go implementation that has patches on top of standard Go to enable integrating BoringCrypto. immudb components are built with this image as a build base.
```
Comment (byo) [inline src/master/production/fips.md:14]: Please check with markdown-lint. 

E.g. here - please add an empty line before unordered list (https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md032---lists-should-be-surrounded-by-blank-lines)

File: src/master/production/fips.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# FIPS 140-2
+
+<WrappedSection>
+
+The Federal Information Processing Standard (FIPS) 140-2 publication describes United States government-approved security requirements for cryptographic modules. [FIPS-140](https://csrc.nist.gov/publications/detail/fips/140/2/final) series is a collection of computer security standards set by the National Institute of Standards and Technology (NIST) for the United States government. FIPS 140–2 defines the critical security parameters vendors must use for encryption before selling to the U.S government.
```
Comment (byo) [inline src/master/production/fips.md:5]:
```suggestion
The Federal Information Processing Standard (FIPS) 140-2 publication describes United States government-approved security requirements for cryptographic modules. [FIPS-140](https://csrc.nist.gov/publications/detail/fips/140/2/final) series is a collection of computer security standards set by the National Institute of Standards and Technology (NIST) for the United States government. FIPS 140–2 defines the critical security parameters vendors must use for encryption before selling their products to the U.S government.
```

File: src/master/production/fips.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+- Currently the builds with FIPS-compliance are only available on `linux-amd64` architecture.
+- There is an overhead in calling into BoringCrypto via cgo for the crypto library functions, which incurs a performance penalty. The library performs slower than the built-in crypto library. Hence you could see a performance drop of ~15% when using a FIPS-compliant immudb server.
+
+</WrappedSection>
+
+<WrappedSection>
+
+### Using FIPS-compliant binaries
+
+You can download the immudb binary from the [latest releases](https://github.com/codenotary/immudb/releases) on Github. The FIPS-compliant binaries have a `-fips` suffix. (Eg: immudb-v1.4.x-Linux-amd64-fips)
```
Comment (byo) [inline src/master/production/fips.md:32]: How the `example gratia` should be written correctly is a bit confusing (see https://painintheenglish.com/case/4723) however I think this is the correct form:

```suggestion
You can download the immudb binary from the [latest releases](https://github.com/codenotary/immudb/releases) on Github. The FIPS-compliant binaries have a `-fips` suffix (e.g. immudb-v1.4.x-Linux-amd64-fips).
```

**Review by byo — APPROVED** [review]
lgtm with some minor comments,

**Review by SimoneLazzaris — APPROVED** [review]

File: src/master/production/fips.md (hunk 4 above)
Comment (arriqaaq) [reply src/master/production/fips.md:14]: Thanks bart for the info, is there some online tool to use for linter that you recommend? I am using this now

https://codebeautify.org/markdown-formatter

File: src/master/production/fips.md (hunk 5 above)
Comment (arriqaaq) [reply src/master/production/fips.md:5]: ack

File: src/master/production/fips.md (hunk 4 above)
Comment (arriqaaq) [reply src/master/production/fips.md:15]: fixed this

Comment (byo) [reply src/master/production/fips.md:15]: I'm using [markdown plugin in VS Code - it has the linter integrated.](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) plugin in VS code it's pretty good there. Was also using [mdl tool](https://github.com/markdownlint/markdownlint) before.

**Review by byo — APPROVED** [review]

**Pull request merged**